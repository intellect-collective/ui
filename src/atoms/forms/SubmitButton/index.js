import React from 'react';
import Button from '../Button';

const SubmitButton = (props) => (
    <Button { ...props } type="submit" />
);
SubmitButton.displayName = 'SubmitButton';
SubmitButton.Wrapped = Button.Wrapped;

export default SubmitButton;
