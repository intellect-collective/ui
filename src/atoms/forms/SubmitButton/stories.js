import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import SubmitButton from './index';

storiesOf('Atoms / Forms / SubmitButton', module)
    .add('default', () => (<SubmitButton name="test-submit-1" onClick={ action('onClick') }>Save</SubmitButton>));
