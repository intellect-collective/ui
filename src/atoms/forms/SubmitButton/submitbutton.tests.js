import React from 'react';
import SubmitButton from '.';
import {
    basicClickableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    type: 'submit',
    onClick: () => {}
};

const expectedProps = {
    name: 'test',
    type: 'submit',
    onClick: expect.any(Function)
};

describe('SubmitButton', () => {
    basicClickableTests(SubmitButton, 'button', validProps, expectedProps);
    // rerenderSuppressionTests(SubmitButton, validProps);

    it('Ignores button type', () => {
        const wrapper = mount(<SubmitButton name="test" type="button" />);
        expect(wrapper.find('button').prop('type')).toEqual('submit');
    });
});
