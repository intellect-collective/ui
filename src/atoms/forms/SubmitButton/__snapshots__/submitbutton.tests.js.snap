// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton Should accept valid props 1`] = `
<SubmitButton
  name="test"
  onClick={[Function]}
  type="submit"
>
  <FormContext
    name="test"
    onClick={[Function]}
    type="submit"
  >
    <Clickable
      name="test"
      onClick={[Function]}
      type="submit"
    >
      <Button
        name="test"
        onClick={[Function]}
        type="submit"
        value=""
      >
        <button
          name="test"
          onClick={[Function]}
          type="submit"
          value=""
        />
      </Button>
    </Clickable>
  </FormContext>
</SubmitButton>
`;
